pybedlite/overlap_detector.py

```diff
@@ -177,85 +180,35 @@ def from_bedrecord(cls: Type["Interval"], record: BedRecord) -> "Interval":
         )
 
 
-_GenericGenomicSpan = TypeVar(
-    "_GenericGenomicSpan",
-    bound=Union[_GenomicSpanWithChrom, _GenomicSpanWithContig, _GenomicSpanWithRefName],
-)
+GenericGenomicsSpan = TypeVar("GenericGenomicsSpan", bound=Union[GenomicSpan, StrandedGenomicSpan])
 """
-A generic genomic feature where the reference sequence name is accessed with any of the 3 most
-common property names ("chrom", "contig", "refname"). This type variable is used for describing the
+A generic genomic feature. This type variable is used for describing the
 generic type contained within the :class:`~pybedlite.overlap_detector.OverlapDetector`.
 """
 
 
-class OverlapDetector(Generic[_GenericGenomicSpan], Iterable[_GenericGenomicSpan]):
+class OverlapDetector(Generic[GenericGenomicsSpan], Iterable[GenericGenomicsSpan]):
```

Comment:
Why is `GenericGenomicSpan` necessary? `StrandedGenomicSpan` subclasses `GenomicSpan`, why not just use `GenomicSpan` here?